.github/workflows/clean_validation_envs.yml

```diff
@@ -0,0 +1,71 @@
+---
+name: Clean Validation Environments
+
+on:
+  schedule:
+    # 2am each night https://crontab.guru/#0_2_*_*_*
+    - cron: "0 2 * * *"
+  workflow_dispatch:
+
+jobs:
+  clean:
+    name: Clean
+    runs-on: ubuntu-latest
+    environment: Dev
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # This is CRITICAL since we're making decisions based on branch existence
+          fetch-depth: 0
+
+      - name: Azure Login
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: clean
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -o errexit
+          set -o pipefail
+          set -o nounset
+          # set -o xtrace
+
+          function deleteEnv ()
+          {
+            local tre_rg="$1"
+            locks=$(az group lock list -g $tre_rg --query [].id -o tsv)
+            # az lock delete --ids ${locks}
+            # az group delete --resource-group "${tre_rg}" --yes --no-wait
+            # each mgmt is per tre so we should delete that too.
+            # az group delete --resource-group "${tre_rg}-mgmt" --yes --no-wait
+          }
+
+          echo "Refs:"
+          git show-ref
+
+          open_prs=$(gh pr list --state open --json number,title)
+
+          # Resource groups that start with a specific string and have the ci_git_ref tag whose value starts with "ref"
+          az group list --query "[?starts_with(name, 'rg-tr') && tags.ci_git_ref != null && starts_with(tags.ci_git_ref, 'refs')].[name, tags.ci_git_ref]" -o tsv |
```

Comment:
isnt the prefix condition too restrictive?
is it possible to have a rg with other prefix?
if so maybe another tag such as 'tre deployment' might be better (instead of checking a prefix)

Reply:
The prefix should be the same one we set in the other workflow where the env gets created. Honestly I could just use the tag but wanted another condition to add some assurence.
The issue with just using tags is that the "main" env is also created in the same process and will get all tags.